pure_pursuit_wrapper/src/pure_pursuit_wrapper.cpp

```diff
@@ -14,42 +14,159 @@
+  RCLCPP_ERROR_STREAM(rclcpp::get_logger("pure_pursuit_wrapper"), "generate_command() cmd.long_accel_mps2: " << cmd.long_accel_mps2);
+  RCLCPP_ERROR_STREAM(rclcpp::get_logger("pure_pursuit_wrapper"), "generate_command() cmd.velocity_mps: " << cmd.velocity_mps);
+  RCLCPP_ERROR_STREAM(rclcpp::get_logger("pure_pursuit_wrapper"), "generate_command() cmd.rear_wheel_angle_rad: " << cmd.rear_wheel_angle_rad);
+  RCLCPP_ERROR_STREAM(rclcpp::get_logger("pure_pursuit_wrapper"), "generate_command() cmd.front_wheel_angle_rad: " << cmd.front_wheel_angle_rad);
+
+  //todo front_wheel_angle_rad is omitted
+  return return_cmd;
+}
+
+autoware_msgs::msg::ControlCommandStamped PurePursuitWrapperNode::generate_command()
+{
+  // process and save the trajectory inside pure_pursuit
+  autoware_msgs::msg::ControlCommandStamped converted_cmd;
+
+  RCLCPP_ERROR_STREAM(rclcpp::get_logger("pure_pursuit_wrapper"), "generate_command() is called");
+
+  if (!current_trajectory_ || !current_pose_ || !current_twist_)
+    return converted_cmd;
+
+
+  motion::control::controller_common::State state_tf = convert_state(current_pose_.get(), current_twist_.get());
+
+  RCLCPP_ERROR_STREAM(rclcpp::get_logger("pure_pursuit_wrapper"), "Forced from frame_id: " << state_tf.header.frame_id << ", into: " << current_trajectory_.get().header.frame_id);
+
+  current_trajectory_.get().header.frame_id = state_tf.header.frame_id;
+
+  process_trajectory_plan(current_trajectory_.get());
+
+  const auto cmd{pp_->compute_command(state_tf)};
+  
+  converted_cmd = convert_cmd(cmd);
+
+
+  return converted_cmd;
+}
+
+rcl_interfaces::msg::SetParametersResult PurePursuitWrapperNode::parameter_update_callback(const std::vector<rclcpp::Parameter> &parameters)
 {
-  plugin_discovery_msg_.name = "pure_pursuit_wrapper_node";
-  plugin_discovery_msg_.version_id = "v1.0";
-  plugin_discovery_msg_.available = true;
-  plugin_discovery_msg_.activated = true;
-  plugin_discovery_msg_.type = cav_msgs::Plugin::CONTROL;
-  plugin_discovery_msg_.capability = "control/trajectory_control";
+  auto error_double = update_params<double>({
+    {"/vehicle_response_lag", config_.vehicle_response_lag}}, parameters);
+  //todo this needs to be updated with new ones, and also update the worker's config parameters while at it
+  rcl_interfaces::msg::SetParametersResult result;
+
+  result.successful = !error_double;
+
+  return result;
 }
 
-bool PurePursuitWrapper::onSpin()
+
+bool PurePursuitWrapperNode::get_availability() 
 {
-  plugin_discovery_pub_(plugin_discovery_msg_);
   return true;
 }
 
+std::string PurePursuitWrapperNode::get_version_id() 
+{
+  return "v1.0";
+}
 
-void PurePursuitWrapper::trajectoryPlanHandler(const cav_msgs::TrajectoryPlan::ConstPtr& tp)
+void PurePursuitWrapperNode::process_trajectory_plan(const carma_planning_msgs::msg::TrajectoryPlan& tp)
```

Comment:
It would be nice if we could make this function available to platooning as well. Maybe move it to basic_autonomy or a new library. Have it return a processed trajectory then pass that to pp.

Reply:
done